### DISCUSSION TITLE
"Failed" mission states still display as succeeded

### DISCUSSION THREAD

**Discussion by ubertpendragon** [FakeFishGames/Barotrauma#15028] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

When setting the state of missions to "Failed" using **MissionStateAction**, the failure text is properly displayed, but the game still shows a positive check mark on the mission
![image](https://github.com/user-attachments/assets/092f3a4c-a958-4293-a16a-e2659ce7269a)

Mission type: abandonedoutpost
`<MissionStateAction missionidentifier="hostagesituation" operation="Set" value="5" />`

https://github.com/FakeFishGames/Barotrauma/blob/7d5b7a310a52372dac0a8bac90fd2a6a2ab342dc/Barotrauma/BarotraumaClient/ClientSource/Events/Missions/AbandonedOutpostMission.cs#L8-L25

Failure states usually don't really work across the board and are hard to find in the code as well

### Reproduction steps

1. Start a Dynamic Europa campaign https://steamcommunity.com/sharedfiles/filedetails/?id=2532991202
2. setreputation coalition 50
3. triggerevent missionevent_hostagesituation
4. Go to the station with the mission
5. Use this event:

```
    <ScriptedEvent identifier="MISSIONSTATEACTIONTEST">
      <MissionStateAction missionidentifier="hostagesituation" operation="Set" value="5" />
    </ScriptedEvent>
```

### Bug prevalence

Happens every time I play

### Single player or multiplayer?

Single player

### Version

v1.6.19.1 (Unto the Breach Update Hotfix 2)

### Which operating system did you encounter this bug on?

Windows

**Comment by Regalis11** [comment]
Addressed in our private repo, fix coming up in the next unstable build and the upcoming release.
